Source/ASDisplayNode.mm

```diff
@@ -2910,11 +2921,11 @@ - (void)didExitHierarchy
   if (ASInterfaceStateIncludesVisible(self.pendingInterfaceState)) {
     void(^exitVisibleInterfaceState)(void) = ^{
       // This block intentionally retains self.
-      __instanceLock__.lock();
-      unsigned isStillInHierarchy = _flags.isInHierarchy;
-      BOOL isVisible = ASInterfaceStateIncludesVisible(_pendingInterfaceState);
-      ASInterfaceState newState = (_pendingInterfaceState & ~ASInterfaceStateVisible);
-      __instanceLock__.unlock();
+      self->__instanceLock__.lock();
+      unsigned isStillInHierarchy = self->_flags.isInHierarchy;
+      BOOL isVisible = ASInterfaceStateIncludesVisible(self->_pendingInterfaceState);
+      ASInterfaceState newState = (self->_pendingInterfaceState & ~ASInterfaceStateVisible);
+      self->__instanceLock__.unlock();
```

Comment:
I don't know that explicit `self` adds much here except visual noise i.e. the self-retain is pretty obvious.

Reply:
The problem is that Xcode / LLVM will throw warnings if you have this setting enabled, e.g. in the ASDKGram example. I think we already resolved these within `PINCache` / `PINRemoteImage` these are the last one that are in Texture.